modules/storages/app/services/storages/file_links/copy_file_links_service.rb

```diff
@@ -111,11 +111,13 @@ def build_location_map(source_files, target_location_map)
         # We need this due to inconsistencies of how we represent the File Path
         target_location_map.transform_keys! { |key| key.starts_with?("/") ? key : "/#{key}" }
 
-        source_files.to_h do |info|
-          target = info.clean_location.gsub(@source.managed_project_folder_path, @target.managed_project_folder_path)
-
-          [info.id.to_s, target_location_map[target]&.id || id]
-        end
+        source_files.filter_map do |info|
+          # in case file has been deleted location is nil
+          if info.location.present?
```

Comment:
:green_circle: Purely cosmetic: This could be written as a guard clause to avoid indentation, i.e.

```ruby
next nil if info.location.blank?
```